File: backend/src/Simulation.java
import java.util.Random;

public class Simulation {
    public static final int STATION_COUNT = 10;
    public Station[] stations;

    public Player p = new Player();
    public Action currentAction = (p, stations) -> {};

    public Simulation(Random r) {
        p.posX = 0;
        p.posY = 0;
        stations = new Station[STATION_COUNT];
        for (int i = 0; i < STATION_COUNT; i++) {
            Station s = new Station();
            s.stationId = i;
            s.lastHarvest = r.nextInt(30, 50);
            s.posX = r.nextInt(-100, 101);
            s.posY = r.nextInt(-100, 101);
            stations[i] = s;
        }
    }

    public void run() {
        currentAction.run(p, stations);
    }

    public String toJSON() {
        return "{\"player\":" + p.toJSON() + ",\"stations\":" + stationJSON() + "}";
    }

    private String stationJSON() {
        String json = "";
        for (int i = 0; i < STATION_COUNT - 1; i++) {
            json += stations[i].toJSON() + ",";
        }
        return "[" + json + stations[STATION_COUNT - 1].toJSON() + "]";
    }
}

File: client-py/simulation.py
class player:
    def __init__(self, pos_x, pos_y, payload, money):
        self.pos_x = pos_x
        self.pos_y = pos_y
        self.payload = payload
        self.money = money

    def __str__(self):
        return f"({self.pos_x},{self.pos_y}) {self.payload} {self.money}"


class station:
    def __init__(self, pos_x, pos_y, last_harvested, station_id):
        self.pos_x = pos_x
        self.pos_y = pos_y
        self.last_harvested = last_harvested
        self.station_id = station_id

    def __str__(self):
        return f"{self.station_id} = ({self.pos_x},{self.pos_y}) {self.last_harvested}"


class simulation:
    def __init__(self, p, stations):
        self.player = p
        self.stations = stations

    def __str__(self):
        stations = '\n\t'.join([str(s) for s in self.stations])
        return f"Simulation:\n\t{str(self.player)}\n\t{stations}"


def create_simulation(json):
    s = json["stations"]
    stations = [station(i["posX"], i["posY"], i["lastHarvested"], i["id"]) for i in s]
    p = json["player"]
    pl = player(p["posX"], p["posY"], p["payload"], p["money"])
    return simulation(pl, stations)
